iotfunctions/db.py

```diff
@@ -570,6 +497,91 @@ def get_column_names(self,table, schema=None):
             table = self.get_table(table, schema)
         
         return [column.key for column in table.columns]
+    
+    
+    def http_request(self, object_type,object_name, request, payload=None, object_name_2=''):
+        '''
+        Make an api call to AS
+        
+        Parameters
+        ----------
+        object_type : str 
+            function,allFunctions, entityType, kpiFunctions
+        object_name : str
+            name of object
+        request : str
+            GET, POST, DELETE, PUT
+        payload : dict
+            Dictionary will be encoded as JSON
+        
+        '''
+        if object_name is None:
+            object_name = ''
+        if payload is None:
+            payload = ''            
+        
+        if self.tenant_id is None:
+            msg = 'tenant_id instance variable is not set. database object was not initialized with valid credentials'
+            raise ValueError(msg)
+        
+        base_url = 'http://%s/api' %(self.credentials['as']['host'])
+        self.url = {}
+        self.url[('allFunctions','GET')] = '/'.join([base_url,'catalog','v1',self.tenant_id,'function?customFunctionsOnly=false'])
+        
+        self.url[('constants','GET')] = '/'.join([base_url,'constants','v1','%s?entityType=%s'%(self.tenant_id,object_name)])
+        self.url[('constants','PUT')] = '/'.join([base_url,'constants','v1','%s?entityType=%s'%(self.tenant_id,object_name)])

```

Comment:
PUT, POST, and DELETE don't use the ?entityType field as a query parameter, they get that information from the constants in the request body

Reply:
Made the change